wordpress-mobile/WordPress-Android
# Hilt migration

This is the parent feature PR for hilt migration. Child PRs: [PR 1](https://github.com/wordpress-mobile/WordPress-Android/pull/16055), [PR 2](https://github.com/wordpress-mobile/WordPress-Android/pull/16151), [PR3](https://github.com/wordpress-mobile/WordPress-Android/pull/16346)

Hilt migration was implemented as an experiment in https://github.com/wordpress-mobile/WordPress-Android/pull/15653 before, but it's a big PR and a bit outdated.
This PR aims to implement Hilt with minimum required changes. Dagger will not be removed completely. Hilt and Dagger will be used together.

To test:
There is no user-facing change. Build the app with different variants to see if it's still buildable and tests are working.

## Regression Notes
1. Potential unintended areas of impact
Some tests might be broken. Debug and release builds might work wrong.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I run current tests. I built the debug app. I relied on automated tests for release build.

3. What automated tests I added (or what prevented me from doing so)
There is no new feature. No tests are added.

PR submission checklist:
